website/content/en/docs/building-operators/golang/project_migration_guide.md

```diff
@@ -1,182 +1,315 @@
+chmod +x setup_envtest.sh
+eval $(./setup_envtest.sh v1.18.2 v3.4.3)
+```
 
-...
-// +kubebuilder:rbac:groups=apps,resources=deployments;pods;daemonsets;replicasets;statefulsets,verbs=get;update;patch;list;create;delete;watch
+**Note:** More info can be found [here][env-test-setup].
 
-func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	reqLogger := log.WithValues("Request.Namespace", req.Namespace, "Request.Name", req.Name)
-	reqLogger.Info("Reconciling Memcached")
+## Migrating your tests
+
+For the new layout you will see that a `controllers/suite_test.go` is created. The project tests are done using [ginkgo](https://onsi.github.io/ginkgo/) and [gomega](https://onsi.github.io/gomega/), look its imports: 
 
+```go
+import (
 ...
-```
-**Note**:
-To update `config/rbac/role.yaml` after changing the markers, run `make manifests`.
-
-The project can now be built, and the operator can be deployed on-cluster. For further steps regarding the deployment of the operator, creation of custom resources and cleaning up of resources, refer to the [quickstart guide][kb_quickstart].
-
-
-[memcached-operator]:/docs/building-operators/golang/quickstart/
-[git_tool]: https://git-scm.com/downloads
-[go_tool]: https://golang.org/dl/
-[docker_tool]:https://docs.docker.com/install/
-[kubectl_tool]: https://github.com/kubernetes/minikube#installation
-[kustomize_tool]: https://github.com/kubernetes-sigs/kustomize/blob/master/docs/INSTALL.md
-[kubebuilder_install]: https://book.kubebuilder.io/quick-start.html#installation
-[memcached_controller]: https://github.com/operator-framework/operator-sdk/blob/8323f56b91590c3bc8098e0024aa825e95386c8a/example/memcached-operator/memcached_controller.go.tmpl
-[sdk_reconcile_struct]: https://github.com/operator-framework/operator-sdk/blob/8323f56b91590c3bc8098e0024aa825e95386c8a/example/memcached-operator/memcached_controller.go.tmpl#L74-L80
-[generate_crd]: https://book.kubebuilder.io/reference/generating-crd.html?highlight=make,mani#generating-crds
-[deployment_yaml]: https://github.com/operator-framework/operator-sdk-samples/blob/master/go/memcached-operator/deploy/operator.yaml
+   	. "github.com/onsi/ginkgo"
+   	. "github.com/onsi/gomega"
+...
+)  
+```                     
+
+SDK has been adopted this stack because it is used in the Kubernetes universe for the projects used by it. To know more about how you can test your controllers see [Writing controller tests][writing-controller-tests]. However, you still able to use the libs that please you to write tests.  
+
+## Migrating your CR's
+
+Now, the CR's are generated in the `/config/samples` so we will need to update them with
+the specs required. In our example, we'll copy the specs from `deploy/crds/cache.example.com_v1alpha1_memcached_cr.yaml` 
+to `config/samples/cache_v1alpha1_memcached.yaml`
```

Comment:
Custom resource samples are stored in `./config/samples` in the new project structure. Copy the examples from your existing project into this directory. In existing projects, CR files have the format `./deploy/crds/<group>.<domain>_<version>_<kind>_cr.yaml`.